MediaBrowser.Api/Playback/MediaInfoService.cs

```diff
@@ -455,72 +461,116 @@ private async Task<PlaybackInfoResponse> GetPlaybackInfo(Guid id, Guid userId, s
-                var streamInfo = string.Equals(item.MediaType, MediaType.Audio, StringComparison.OrdinalIgnoreCase) ?
-                    streamBuilder.BuildAudioItem(options) :
-                    streamBuilder.BuildVideoItem(options);
+                    // The MediaSource supports direct stream, now test to see if the client supports it
+                    var streamInfo = string.Equals(item.MediaType, MediaType.Audio, StringComparison.OrdinalIgnoreCase) ?
+                        streamBuilder.BuildAudioItem(options) :
+                        streamBuilder.BuildVideoItem(options);
 
-                if (streamInfo == null || !streamInfo.IsDirectStream)
-                {
-                    mediaSource.SupportsDirectStream = false;
-                }
+                    if (streamInfo == null || !streamInfo.IsDirectStream)
+                    {
+                        mediaSource.SupportsDirectStream = false;
+                    }
 
-                if (streamInfo != null)
-                {
-                    SetDeviceSpecificSubtitleInfo(streamInfo, mediaSource, auth.Token);
-                }
+                    if (streamInfo != null)
+                    {
+                        SetDeviceSpecificSubtitleInfo(streamInfo, mediaSource, auth.Token);
+                    }
+				}
             }
 
             if (mediaSource.SupportsTranscoding)
             {
-                options.MaxBitrate = GetMaxBitrate(maxBitrate, user);
-
+				if (mediaSource.IsRemote && user.Policy.ForceRemoteSourceTranscoding)
+				{
+					if (GetMaxBitrate(maxBitrate, user) < mediaSource.Bitrate)
+					{
+						options.MaxBitrate = GetMaxBitrate(maxBitrate, user);
+					}
+					else
+					{
+						options.MaxBitrate = mediaSource.Bitrate;
+					}
+				}
+				else
+			    {
+					options.MaxBitrate = GetMaxBitrate(maxBitrate, user);
+				}
+				
                 // The MediaSource supports direct stream, now test to see if the client supports it
                 var streamInfo = string.Equals(item.MediaType, MediaType.Audio, StringComparison.OrdinalIgnoreCase) ?
                     streamBuilder.BuildAudioItem(options) :
                     streamBuilder.BuildVideoItem(options);
 
-                if (streamInfo != null)
-                {
-                    streamInfo.PlaySessionId = playSessionId;
-
-                    if (streamInfo.PlayMethod == PlayMethod.Transcode)
+				if (mediaSource.IsRemote && user.Policy.ForceRemoteSourceTranscoding)
+				{
+                    if (streamInfo != null)
                     {
+                        streamInfo.PlaySessionId = playSessionId;    
                         streamInfo.StartPositionTicks = startTimeTicks;
                         mediaSource.TranscodingUrl = streamInfo.ToUrl("-", auth.Token).TrimStart('-');
+                        mediaSource.TranscodingUrl += "&allowVideoStreamCopy=false";
+                        mediaSource.TranscodingUrl += "&allowAudioStreamCopy=false";
```

Comment:
this doesn't feel as a correct way of composing a URL... I now see it's a copy from down below, so my question is - why same thing is done twice?

Reply:
the composed URL method is a continuation of the original code. I think this might section might be the issue I was having where it would force transcode everything instead of doing a simple remux.

Reply:
We sure that we must disallow audio stream copy? Shouldn't it be based on `allowAudioStreamCopy`? If the audio stream is already supported and we allow copy, why transcode it?